### PR TITLE
ROC-4207 Save claimant response send email

### PR DESCRIPTION
**Pull request by timja**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-4207

### Change description ###
Send claimant response saved email

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/events/claimantresponse/ClaimantResponseActionsHandler.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        NotificationsProperties notificationsProperties
+    ) {
+        this.notificationService = notificationService;
+        this.notificationsProperties = notificationsProperties;
+    }
+
+    @EventListener
+    public void sendNotificationToDefendant(ClaimantResponseEvent event) {
+        Claim claim = event.getClaim();
+        this.notificationService.sendMail(
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/events/claimantresponse/ClaimantResponseActionsHandler.java]: I think we can reuse `NotificationToDefendantService` to send this email, as code below is exactly sending the same email template.

Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/events/claimantresponse/ClaimantResponseActionsHandler.java]: I agree with you Sabah, however if you have to change the template, then it need to reworked as now.  if Tim is changing for this, then he need to refactor all places even in ccd.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/NotificationReferenceBuilder.java [hunk 2]
```diff
@@ -165,4 +165,17 @@ public static String referenceForDefendant(String claimReferenceNumber, String o
             return reference(TEMPLATE, DEFENDANT, otherParty.toLowerCase(), claimReferenceNumber);
         }
     }
+
+    public static class ClaimantResponse {
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/NotificationReferenceBuilder.java]: We don't need this new class as we already have similar static inner class in same file called `ClaimantResponseSubmitted`.

Comment (timja) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/NotificationReferenceBuilder.java]: its gone now

File: src/main/java/uk/gov/hmcts/cmc/claimstore/events/claimantresponse/ClaimantResponseActionsHandler.java (hunk 1 above)
Comment (timja) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/events/claimantresponse/ClaimantResponseActionsHandler.java]: done, didn't see that before

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/events/claimantresponse/ClaimantResponseActionsHandler.java]: Thanks @timja 
 @dharmendrak  `CCJ` cleanup can be done in separate PR IMO

**Pull request merged**